library/mobile/api_store.go

```diff
@@ -0,0 +1,17 @@
+package gowaku
+
+import (
+	"github.com/waku-org/go-waku/library"
+)
+
+// StoreQuery is used to retrieve historic messages using waku store protocol.
+func StoreQuery(queryJSON string, peerID string, ms int) string {
+	response, err := library.StoreQuery(queryJSON, peerID, ms)
+	return prepareJSONResponse(response, err)
+}
+
+// StoreLocalQuery is used to retrieve historic messages stored in the localDB using waku store protocol.
+func StoreLocalQuery(queryJSON string) string {
```

Comment:
Why are 2 different methods required? 
I am thinking, wouldn't it be better to expose to the user a simple method called StoreQuery which checks localDB and if not present fetches from the peer?

Reply:
The problem is that you do not know if the data you have locally is complete. For an specific time period you might have `n` messages on your db, and the peer might have `n+m` message, so attempting to retrieve messages locally might return results and still end up missing messages. :thinking:

Reply:
Ok, it's the users responsibility to identify what is missing and query peer accordingly.